evm/src/cpu/kernel/tests/account_code.rs

```diff
@@ -52,19 +51,18 @@ fn prepare_interpreter(
     interpreter.run()?;
     assert_eq!(interpreter.stack(), vec![]);
 
-    let k = nibbles_64(U256::from_big_endian(
-        keccak(address.to_fixed_bytes()).as_bytes(),
-    ));
-    // Next, execute mpt_insert_state_trie.
-    interpreter.generation_state.registers.program_counter = mpt_insert_state_trie;
+    let k = keccak(address.to_fixed_bytes());
+    // Next, execute smt_insert_state.
+    interpreter.generation_state.registers.program_counter = smt_insert_state;
     let trie_data = interpreter.get_trie_data_mut();
     if trie_data.is_empty() {
-        // In the assembly we skip over 0, knowing trie_data[0] = 0 by default.
+        // In the assembly we skip over 0, knowing trie_data[0:4] = 0 by default.
```

Comment:
I might be missing something, but isn't it only the first 2 elements that are set to 0 by default? (Looking at `load_state_smt`, it seems that we get the trie data size, add 2 for an empty hash node, and then store the trie root)

Reply:
Ah this is because I wasn't calling `interpreter.generation_state.state_smt_prover_inputs = state_smt_prover_inputs_reversed(&trie_inputs);`. I've added it and now I don't need to manually extend the trie data.